turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -1553,110 +1528,111 @@ impl FileSystemPath {
+        if !matches!(*current.get_type().await?, FileSystemEntryType::Symlink) {
+            return Ok(RealPathResult {
+                path: current,
+                symlinks: symlinks.into_iter().collect(), // convert set to vec
             }
+            .cell());
+        }
 
-            if let LinkContent::Link { target, link_type } = &*current_vc.read_link().await? {
-                symlinks.insert(current_vc);
-                current_vc = if link_type.contains(LinkType::ABSOLUTE) {
-                    current_vc.root()
-                } else {
-                    *parent_result.path
-                }
-                .join(target.clone())
-                .to_resolved()
-                .await?;
+        if let LinkContent::Link { target, link_type } = &*current.read_link().await? {
+            symlinks.insert(current.clone());
+            current = if link_type.contains(LinkType::ABSOLUTE) {
+                (*current.root().await?).clone()
             } else {
-                // get_type() and read_link() might disagree temporarily due to turbo-tasks
-                // eventual consistency or if the file gets invalidated before the directory does
-                return Ok(RealPathResult {
-                    path: current_vc,
-                    symlinks: symlinks.into_iter().collect(), // convert set to vec
-                }
-                .cell());
+                parent_result.path
+            }
+            .join(target.clone())?;
+        } else {
+            // get_type() and read_link() might disagree temporarily due to turbo-tasks
+            // eventual consistency or if the file gets invalidated before the directory does
+            return Ok(RealPathResult {
+                path: current,
+                symlinks: symlinks.into_iter().collect(), // convert set to vec
             }
+            .cell());
         }
+    }
 
-        // Too many attempts or detected a cycle, we bailed out!
-        //
-        // TODO: There's no proper way to indicate an non-turbo-tasks error here, so just return the
-        // original path and all the symlinks we followed.
-        //
-        // Returning the followed symlinks is still important, even if there is an error! Otherwise
-        // we may never notice if the symlink loop is fixed.
-        Ok(RealPathResult {
-            path: self,
-            symlinks: symlinks.into_iter().collect(),
-        }
-        .cell())
+    // Too many attempts or detected a cycle, we bailed out!
+    //
+    // TODO: There's no proper way to indicate an non-turbo-tasks error here, so just return the
+    // original path and all the symlinks we followed.
+    //
+    // Returning the followed symlinks is still important, even if there is an error! Otherwise
+    // we may never notice if the symlink loop is fixed.
+    Ok(RealPathResult {
+        path: current,
+        symlinks: symlinks.into_iter().collect(),
     }
+    .cell())
 }
 
 #[turbo_tasks::value_impl]
```

Comment:
It seems like the difference of `ValueToString` and `Display` implementation of `FileSystemPath` is the cause of the CI failure.